validator/client/validator.go

```diff
@@ -486,6 +469,54 @@ func (v *validator) domainData(ctx context.Context, epoch uint64, domain []byte)
 	return res, nil
 }
 
+func (v *validator) logDuties(slot uint64, duties []*ethpb.DutiesResponse_Duty) {
+	attesterKeys := make([][]string, params.BeaconConfig().SlotsPerEpoch)
+	for i := range attesterKeys {
+		attesterKeys[i] = make([]string, 0)
+	}
+	proposerKeys := make([]string, params.BeaconConfig().SlotsPerEpoch)
+	slotOffset := (slot / params.BeaconConfig().SlotsPerEpoch) * params.BeaconConfig().SlotsPerEpoch
```

Comment:
I dont quite understand  `(slot / params.BeaconConfig().SlotsPerEpoch) * params.BeaconConfig().SlotsPerEpoch` 

Do you mean to do `slot - epoch_start_slot`?

Reply:
Yes; slot at the beginning of this slot's epoch.  Is there a helper for it?

Reply:
helpers.StartSlot()

Reply:
Thanks; updated to use the helpers.